mmtrack/models/track_heads/stark_head.py

```diff
@@ -0,0 +1,376 @@
+        return out_dec, memory
+
+
+@HEADS.register_module()
+class StarkHead(DETRHead):
+    """
+    Args:
+        num_classes (int): Number of categories excluding the background.
+        in_channels (int): Number of channels in the input feature map.
+        num_query (int): Number of query in Transformer.
+        num_cls_fcs (int, optional): Number of fully-connected layers used in
+            score head. Default to 3.
+        stride (int): The stride of the input feature map. Default to 16.
+        transformer (obj:`mmcv.ConfigDict`|dict): Config for transformer.
+            Default: None.
+        positional_encoding (obj:`mmcv.ConfigDict`|dict):
+            Config for position encoding.
+        loss_cls (obj:`mmcv.ConfigDict`|dict): Config of the
+            classification loss. Default `CrossEntropyLoss`.
+        loss_bbox (obj:`mmcv.ConfigDict`|dict): Config of the
+            regression loss. Default `L1Loss`.
+        loss_iou (obj:`mmcv.ConfigDict`|dict): Config of the
+            regression iou loss. Default `GIoULoss`.
+        tran_cfg (obj:`mmcv.ConfigDict`|dict): Training config of
+            transformer head.
+        test_cfg (obj:`mmcv.ConfigDict`|dict): Testing config of
+            transformer head.
+        init_cfg (dict or list[dict], optional): Initialization config dict.
+            Default: None
+    """
+    def __init__(self,
+                 num_classes,
+                 in_channels,
+                 num_query=1,
+                 num_cls_fcs=3,
+                 stride=16,
+                 transformer=None,
+                 positional_encoding=dict(
+                     type='SinePositionalEncoding',
+                     num_feats=128,
+                     normalize=True),
+                 loss_cls=dict(
+                     type='CrossEntropyLoss',
+                     use_sigmoid=False,
+                     loss_weight=1.0,
+                 ),
+                 loss_bbox=dict(type='L1Loss', loss_weight=5.0),
+                 loss_iou=dict(type='GIoULoss', loss_weight=2.0),
+                 train_cfg=None,
+                 test_cfg=dict(max_per_img=100),
+                 init_cfg=None,
+                 **kwargs):
+        self.num_cls_fcs = num_cls_fcs
+        self.stride = stride
+        super(StarkHead, self).__init__(
+            num_classes,
+            in_channels,
+            num_query=num_query,
+            transformer=transformer,
+            positional_encoding=positional_encoding,
+            loss_cls=loss_cls,
+            train_cfg=train_cfg,
+            test_cfg=test_cfg,
+            init_cfg=init_cfg,
+        )
+
+    def _init_layers(self):
+        feat_size = self.test_cfg['search_size'] // self.stride
+        self.bbox_head = CornerPredictorHead(
+            self.embed_dims,
+            self.embed_dims,
+            feat_size=feat_size,
+            stride=self.stride)
+        self.cls_head = ScoreHead(self.embed_dims, self.embed_dims, 1,
```

Comment:
for the names of `CornerPredictorHead` and `ScoreHead`, do we have names mentioned in the paper?

Reply:
not explicitly.   mention "bbox prediction head" and "score prediction head"